samples/stored-procedures/clear.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * A stored procedure for Azure DocumentDB which deletes documents with a specified filter key and value
+ * @function
+ * @param  {String} filterOn = 'type'         The key to filter documents on for deletion.
+ * @param  {String} filterValue = 'session'   The value that a document's filter key must have to be deleted
+ * @return {responseBody}
+ */
+function clear(filterOn, filterValue) {
+
+  // set default filter key and value
+  filterOn = filterOn || 'type';
+  filterValue = filterValue || 'session';
+
+  const response = __.response; // get the response object
+
+  /**
+   * The response body returned by the stored procedure
+   * @const
+   * @typedef {Object} responseBody
+   * @type {Object} responseBody
+   * @prop {Number} deleted        The number of documents which were deleted.
+   * @prop {Boolean} continuation  Whether there are still more documents to delete.
+   */
+  const responseBody = {
+    deleted: 0,
+    continuation: true,
+  };
+
+  /**
+   * Recursively deletes each session in an array, and then attempts to get more to delete
+   * @function
+   * @param  {Array} sessions  The array of session documents to delete

```

Comment:
can we rename sessions for function name and function parameter to more generic? Like deleteDocuments etc.